text/0037-minting.md

```diff
@@ -0,0 +1,167 @@
+
+## Governors: The Signer Set for a Consensus Validator
+[governors]: #governors
+
+The consensus validator nodes must maintain a set of "master minters," also known as _governors_, who can sign the `MintConfigTx` transactions. The governor set is maintained in the node's startup configuration file, and it is the responsibility of the node operator to perform due diligence on any governor they are adding to the set.
+
+The governor set is a set of Ed25519 public keys and a threshold, and a `MintConfigTx`'s `TxPrefix` must be signed by the threshold number of governors required by each node in a quorum of nodes to establish a new Minting Configuration for the network.
+
+To further protect the governors configuration, and to ensure all nodes are running with identical configuration, the consensus enclave contains a baked-in Ed25519 public key called the `Minting Trust Root` key. The consensus enclave requires that the governors configuration be signed by the matching private key (held by the MobileCoin Foundation), and that the signature is provided upon initialization. Without a valid signature over the governors configuration, the consensus enclave will refuse to initialize.
+
+The governors configuration is also hashed into the node's responder ID, preventing node-to-node attestation if the configuration differs. This is similar to how minimum fee configuration is currently handled.
+
+### Multiple Asset Fees
+
+The startup-config for the consensus validator node contains the governors, as well as a `TokensConfig`, which specifies the following per token:
+
+* Token ID
+* Minimum fee
+* Whether any fee is allowed 
+* The list of governors who can change the Minting Config for this token (issue `MintConfigTx` transactions)
+
+Every transaction's fee is paid in the same token type as the inputs and outputs of the transaction.
+
+Prioritization on the consensus validator nodes is determined from the fee of the transaction, multiplied by a priority number derived from the minimum fee for the token asset (see the [Tx fee priority commit](https://github.com/mobilecoinfoundation/mobilecoin/commit/9450b1966cef23cedd53cac5f169110019fe2f2d). Therefore, prioritization can occur without revealing the token types involved in the transactions in a block.
+
+A client can adjust its fee during times of congestion by knowing both the minimum fee for the token type they are attempting to transact, along with the rejected fee value they submitted.
+
+Currently, the enclaves aggregate the fee for multiple transactions in the same block so that there is only one fee output in order to save space in the ledger. With multiple fees, the aggregation will produce a number of fee outputs capped by the minimum of either the number of token IDs in the block or the number of transactions in a block.
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Adding multiple confidential asset types with minting, burning, and bridge operation increases the complexity of the system.
```

Comment:
It feels like some of this section could address some of the downsides of the design. Like what's the downside of having a list approved governors? What's the downside of having maybe poorly configured tokens on chain?

Reply:
I am not sure there are any obvious downsides to this. It adds complexity but it is necessary since we do need some kind of a mechanism to limit who can mint and what can be minted.
I am not sure poorly configured tokens is a thing. If we configure minting of a token to have relaxed minting requirements (e.g. only 1 signer for example), whether that is poorly configured or not depends on what we are trying to achieve. For a test token that might be perfectly fine.